File: backend/src/main/java/com/NBE3_4_2_Team4/domain/product/product/dto/ProductResponseDto.java
package com.NBE3_4_2_Team4.domain.product.product.dto;

import com.NBE3_4_2_Team4.domain.product.product.entity.Product;
import com.NBE3_4_2_Team4.domain.product.saleState.entity.SaleState;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

public class ProductResponseDto {

    @Builder
    @Getter
    @NoArgsConstructor
    @AllArgsConstructor
    public static class GetItems {

        private Long productId;

        private String productName;

        private int productPrice;

        private String productDescription;

        private String productImageUrl;

        private String productCategory;

        private String productSaleState;

        public GetItems(Product product, String productCategory, SaleState productSaleState) {
            this.productId = product.getId();
            this.productName = product.getName();
            this.productPrice = product.getPrice();
            this.productDescription = product.getDescription();
            this.productImageUrl = product.getImageUrl();
            this.productCategory = productCategory;
            this.productSaleState = productSaleState.name();
        }
    }

    @Builder
    public record GetItemsByKeyword(String keyword, List<GetItems> products) {

    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/product/product/dto/ProductResponseDto.kt
package com.NBE3_4_2_Team4.domain.product.product.dto

import com.NBE3_4_2_Team4.domain.product.product.entity.Product
import com.NBE3_4_2_Team4.domain.product.saleState.entity.SaleState
import com.NBE3_4_2_Team4.standard.dto.PageDto
import org.springframework.data.domain.Page

class ProductResponseDto {

    data class GetItem(

        val productId: Long,

        val productName: String,

        val productPrice: Int,

        val productDescription: String?,

        val productImageUrl: String,

        val productCategory: String,

        val productSaleState: String
    ) {
        constructor(product: Product, productCategory: String, productSaleState: SaleState) : this(
            productId = product.id!!,
            productName = product.name,
            productPrice = product.price,
            productDescription = product.description,
            productImageUrl = product.imageUrl,
            productCategory = productCategory,
            productSaleState = productSaleState.name
        )
    }

    data class GetItemsByKeyword(
        val keyword: String,
        val products: List<GetItem>
    )

    class PageDtoWithKeyword<T>(
        page: Page<T>,
        val keyword: String
    ) : PageDto<T>(page)
}